Compiler Errors on Apple Clang 13.1.6

```
../wpiutil/src/main/native/include/wpi/future.h:337:52: error: 'result_of<(lambda at ../wpiutil/src/test/native/cpp/future_test.cpp:16:35) &&(bool &&)>' is deprecated [-Werror,-Wdeprecated-declarations]
  template <typename F, typename R = typename std::result_of<F && (T &&)>::type>
                                                   ^
```

```
../wpinet/src/main/native/thirdparty/libuv/src/unix/darwin.cpp:214:17: error: variable 'kr' set but not used [-Werror,-Wunused-but-set-variable]
  kern_return_t kr;
```

--- a/wpiutil/src/main/native/include/wpi/future.h
+++ b/wpiutil/src/main/native/include/wpi/future.h
@@ -334,7 +334,7 @@ class future final {
     }
   }
 
-  template <typename F, typename R = typename std::result_of<F && (T &&)>::type>
+  template <typename F, typename R = typename std::invoke_result_t<F&&, T&&>>
   future<R> then(F&& func) {
     return then(PromiseFactory<R>::GetInstance(), std::forward<F>(func));
   }
@@ -457,7 +457,7 @@ class future<void> final {
     }
   }
 
-  template <typename F, typename R = typename std::result_of<F && ()>::type>
+  template <typename F, typename R = typename std::invoke_result_t<F&&>>
   future<R> then(F&& func) {
     return then(PromiseFactory<R>::GetInstance(), std::forward<F>(func));
   }

